containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -350,8 +354,8 @@ class UnorderedMap {
     /// properties.
     const auto prop_copy =
         Impl::with_properties_if_unset(arg_prop, std::string("UnorderedMap"));
-    const auto prop_copy_noinit =
-        Impl::with_properties_if_unset(prop_copy, Kokkos::WithoutInitializing);
+    const auto prop_copy_noinit = Impl::with_properties_if_unset_and_compatible(
```

Comment:
I would handle the incompatibility here directly. It's not clear what what effect this function call has and if we need to provide a comment we might just as well move the code here without further changes to `Kokkos_ViewCtor.hpp`.